src/components/capture/creation/index.tsx

```diff
@@ -136,20 +148,24 @@ function NewCaptureModal() {
 
             setFormSubmitting(true);
 
-            handlers.close();
+            exitModal();
         },
 
         close: () => {
             if (hasChanges()) {
-                console.log('uh oh');
-            }
-            if (schemaFromEditor) {
-                removeSchema();
+                confirmationModalContext
+                    ?.showConfirmation({
+                        message: 'confirm.loseData',
+                    })
+                    .then((confirmed) => {
+                        if (confirmed) {
+                            exitModal();
+                        }
+                    })
+                    .catch(() => {});
+            } else {
+                exitModal();
```

Comment:
If there are changes we now want to make sure they mean to close. 

If the modal explodes somehow and they get into the `catch` path I _think_ keeping them in the modal is correct. Might be annoying to be stuck there but better than losing work on accident.